glacier_mapping/train.py

```diff
@@ -136,13 +145,30 @@ def log_images(writer, frame, batch, epoch, stage="train"):
     Return:
         Images Logged onto tensorboard
     """
-    pm = lambda x: x.permute(0, 3, 2, 1)
+    pm = lambda x: x.permute(0, 3, 1, 2)
     squash = lambda x: (x - x.min()) / (x.max() - x.min())
 
     x, y = batch
-    y_hat = torch.sigmoid(frame.infer(x))
+    if frame.multi_class:
```

Comment:
This is much less readable than the previous version of train.

I think you could modularize it. Alternatively, I personally don't think we need to support both 1 and multichannel outputs (we can just have multichannel). But should check with others to see if they think it's okay.

Reply:
it can be changed to something like frame.pred, it might cause a little duplication though. I changed it a little, you can have a look at it or we can support only 2 classes.